src/port1.0/portbump.tcl

```diff
@@ -0,0 +1,316 @@
+                }
+            }
+
+            # Append the string for any of our default types that were
+            # not previously calculated
+            if {[llength $missing_types]} {
+                # get the full path of the distfile.
+                set fullpath [file join $distpath $distfile]
+                if {![file isfile $fullpath]} {
+                    return -code error "$distfile does not exist in $distpath"
+                }
+                # foreach type $missing_types {
+                #     lappend both_checksums $type "-" [portchecksum::calc_$type $fullpath]
+                # }
+            }
+        }
+
+        global version revision
+
+        ui_msg "We will bump these:"
+        foreach {type sum calculated_sum} $both_checksums {
+            ui_msg [format "Old %-8s %s" ${type}: $sum]
+            ui_msg [format "New %-8s %s" ${type}: $calculated_sum]
+        }
+
+        if {[info exists macports::ui_options(questions_yesno)]} {
+            set retval [$macports::ui_options(questions_yesno) $msg "BumpPrompt" "" {y} 0 "Do you approve of these changes?"]
+            if {$retval == 0} {
+                # User said yes
+                # TODO: Ask to reset revision number only if > 0 and pass revision number to message.
+                set retval [$macports::ui_options(questions_yesno) $msg "BumpPrompt" "" {y} 0 [format "Do you want to reset revision number (currently %s) to 0?" $revision]]
+                if {$retval == 0} {
+                    # User said yes
+                    # TODO: Reset revision to 0
+                }
+            } else {
+                # User said no
+                return 0
+            }
+        }
+
+        set patterns {}
+
+        set whitespace {[[:space:]\\]+}
+        # Create substitution pattern(s) for checksum
+        foreach {type sum calculated_sum} $both_checksums {
+            lappend patterns "s/(${type}${whitespace})${sum}/\\1${calculated_sum}/g"
+        }
+        # Create substitution pattern for revision (reset to 0)
+        lappend patterns {s/(revision[[:space:]\\]+)[0-9]+/\10/g}
+
+        # Construct sed command
+        set cmdline {}
+        lappend cmdline $portutil::autoconf::sed_command -E
+        foreach {pattern} $patterns {
+            lappend cmdline -e $pattern
+        }
+
+        # TODO: Implement patch use case
+        # Reference: https://guide.macports.org/chunked/development.patches.html
+
+        # Get the uid of Portfile owner
+        set owneruid [name_to_uid [file attributes ${portfile} -owner]]
+
+        # root -> owner id
+        exec_as_uid $owneruid {
+            # Create temporary Portfile.bump.XXXXXX
+            if {[catch {set tmpfile [mkstemp "${portpath}/Portfile.bump.XXXXXX"]} error]} {
+                ui_debug $::errorInfo
+                ui_error "mkstemp: $error"
+                return -code error "mkstemp failed"
+            } else {
+                # Extract the Tcl Channel number
```

Comment:
Since you're erroring out in the if block, there's no need to put this code in an else block.

Reply:
Removed the else